### ISSUE TITLE
implement dark mode

### ISSUE THREAD

**Issue by dm161** [golang-cafe/golang.cafe#42]
styles are inlined/internal only. To keep latency low and avoid any render blocking resource, we might need to keep using internal css, but there is a way to re-use and import common css by using Go nested templates.

This can be achieved easily as follows:

**common-css.html**
```
{{ define "common-css" }}
<style type="text/css">
// common css code here ...
</style>
{{ end }}
```

**landing.html**
```
<head:>
...
{{ template "common-css" }}
</head>
...
```

Please see discussions and issues
- https://github.com/golang-cafe/golang.cafe/pull/39
- https://github.com/golang-cafe/golang.cafe/issues/36

**Comment by vitaly-zdanevich** [comment]
For me this is already possible to start working on this issue or I need to wait to some refactor to finish?

**Comment by dm161** [comment]
I am not aware of anybody working on the nested template refactoring yet, so If you want you can give it a go :)

I am not sure how you want to implement it? I guess some sort of button to switch on and off dark mode? Open to your suggestions

**Comment by vitaly-zdanevich** [comment]
> how you want to implement it?

https://developer.mozilla.org/en-US/docs/Web/CSS/@media/prefers-color-scheme

**Comment by dm161** [comment]
I think it will be pretty straightforward then, you can either copy pasta all the dark mode css everywhere or create the `dark-mode-css.html` nested template as shown above so it can be re-used. Up to how best works for you :)

**Comment by vitaly-zdanevich** [comment]
Thanks. How can I check my changes locally without installing Docker, PostgreSQL?

**Comment by dm161** [comment]
Right now you must need the following in order to run it locally: Bash, Go, Docker. You can just run a script to set it up. `./setup-database.sh` - are you having issues with Docker? You don't need to install PostgreSQL locally you just need docker running and the bash script will do the rest

**Comment by vitaly-zdanevich** [comment]
Do not want to pollute my system with Docker, I will try with Devtools only :)

**Comment by dm161** [comment]
fair enough - not sure if there are other ways you can test this locally. Docker is a pretty standard tool for dev work. Can use postgres installed locally as well. But up to you really. Open to suggestions for better dev experience if you have any
